src/components/connection-modal/update-peripheral-step.jsx

```diff
@@ -0,0 +1,262 @@
+                    </Box>
+                    <img
+                        className={styles.helpStepImage}
+                        src={this.props.connectionSmallIconURL}
+                    />
+                    {
+                        // The instructions for getting the peripheral ready for the update process will vary
+                        // depending on the peripheral. Should we get this from the extension somehow?
+                    }
+                    <FormattedMessage
+                        className={styles.helpStepText}
+                        defaultMessage="Connect your {extensionName} to this device using a USB cable."
+                        description="Instructions to connect the micro:bit to the computer for the update process"
+                        id="gui.connection.updatePeripheral.microBitConnect"
+                        values={{
+                            extensionName: this.props.name
+                        }}
+                    />
+                </Box>
+                <Box className={styles.scratchLinkHelpStep}>
+                    <Box className={styles.helpStepNumber}>
+                        {'2'}
+                    </Box>
+                    <img
+                        className={styles.helpStepImage}
+                        src={sendUpdateIcon}
+                    />
+                    <FormattedMessage
+                        defaultMessage="Press &quot;Do Update&quot; and allow the update to complete."
+                        description="Instructions to press the button to begin the update process"
+                        id="gui.connection.updatePeripheral.pressUpdate"
+                    />
+                </Box>
+            </Box>
+        </Box>);
+    }
+
+    renderSendUpdate () {
+        return (<Box className={styles.activityArea}>
+            <ProgressRingComponent
+                sizePx={36}
+                value={this.state.progressPercentage}
+                max={100}
+            />
+            <FormattedMessage
+                defaultMessage="Updating {progressPercentage}%"
+                description="Progress message while updating the peripheral"
+                id="gui.connection.updatePeripheral.progress"
+                values={{
+                    progressPercentage: Math.floor(this.state.progressPercentage)
+                }}
+            />
+        </Box>);
+    }
+
+    renderResults () {
+        let resultsContent;
+        if (this.state.err === null) {
+            resultsContent = (<FormattedMessage
+                defaultMessage="Update successful!"
+                description="Message to indicate that the peripheral update was successful"
+                id="gui.connection.updatePeripheral.updateSuccessful"
+            />);
+        } else if (this.state.err.message === 'No valid interfaces found.') {
+            // this is a special case where the micro:bit's communication firmware is too old to support WebUSB
```

Comment:
FWIW, I hit this step when I had taken one of our Micro:bit 1's and installed a MakeCode hex file onto it. (I guess the issue was the underlying firmware version, not the MakeCode hex)

Reply:
Yep, I have a v1 micro:bit that I've intentionally kept on the old communications firmware just to test this :)
I don't believe this can happen with a v2